src/XR/native/nativeXRFrame.ts

```diff
@@ -0,0 +1,92 @@
+import { RegisterNativeTypeAsync } from "../../Engines/nativeEngine";
+
+/** @hidden */
+interface INativeXRFrame extends XRFrame {
+    // Native-only helper functions
+    getPoseData: (space: XRSpace, baseSpace: XRReferenceSpace, vectorBuffer: ArrayBuffer, matrixBuffer: ArrayBuffer) => XRPose;
+}
+
+/** @hidden */
+export class NativeXRFrame implements XRFrame {
+    private readonly _xrTransform = new XRRigidTransform();
+    private readonly _xrPose: XRPose = {
+        transform: this._xrTransform,
+        emulatedPosition: false
+    };
+    // Enough space for position, orientation
+    private readonly _xrPoseVectorData = new Float32Array(4 * (4 + 4));
+
+    public get session(): XRSession {
+        return this._nativeImpl.session;
+    }
+
+    constructor(private _nativeImpl: INativeXRFrame) { }
+
+    public getPose(space: XRSpace, baseSpace: XRReferenceSpace): XRPose | undefined {
+        if (!this._nativeImpl.getPoseData(space, baseSpace, this._xrPoseVectorData.buffer, this._xrTransform.matrix.buffer)) {
+            return undefined;
+        }
+        const position = this._xrTransform.position as DOMPoint;
+        position.x = this._xrPoseVectorData[0];
+        position.y = this._xrPoseVectorData[1];
+        position.z = this._xrPoseVectorData[2];
+        position.w = this._xrPoseVectorData[3];
+
+        const orientation = this._xrTransform.orientation as DOMPoint;
+        orientation.x = this._xrPoseVectorData[4];
+        orientation.y = this._xrPoseVectorData[5];
+        orientation.z = this._xrPoseVectorData[6];
+        orientation.w = this._xrPoseVectorData[7];
+        return this._xrPose;
+    }
+
+    public fillPoses(spaces: XRSpace[], baseSpace: XRSpace, transforms: Float32Array): boolean {
+        return this._nativeImpl.fillPoses!(spaces, baseSpace, transforms);
+    }
+
+    public getViewerPose(referenceSpace: XRReferenceSpace): XRViewerPose | undefined {
+        return this._nativeImpl.getViewerPose(referenceSpace);
+    }

```

Comment:
I don't know if you intend on wrapping all of these functions, but if you just want to pass through, I believe you can just assign the function instead of forwarding. Saves one function call.

```suggestion
    public readonly getViewerPose = this._nativeImpl.getViewerPose;
```